Example/Tests/UserTests.swift

```diff
@@ -215,6 +215,50 @@ class UserTests: QuickSpec {
             }
         }
 
+        describe("Keychain management") {
+            it("Should not treat the user as logged in if it does not find keychain data") {
+                let user = User(state: .loggedOut)
+                expect(user.state).to(equal(UserState.loggedOut))
+            }
+
+            it("Should treat the user as logged in if it finds keychain data") {
+                Utils.createDummyKeychain()
+                let user = User(state: .loggedOut)
+                try? user.loadStoredTokens()
+                expect(user.state).to(equal(UserState.loggedIn))
+            }
+
+            it("Should update keychain when user is refreshed") {
+                self.stub(uri("/oauth/token"), try! Builders.load(file: "valid-refresh", status: 200))
+                var newTokens: TokenData?
+                do {
+                    Utils.createDummyKeychain()
+                    let user = User(state: .loggedOut)
+                    try? user.loadStoredTokens()
+                    user.refresh { _ in }
```

Comment:
`User.refresh` is asynchronous so I guess this could potentially fail `TestingUser` implements the same methods as `User` but synchronously instead.